server/src/fstar_connection.ts

```diff
@@ -0,0 +1,512 @@
+import {
+	URI
+} from 'vscode-uri';
+
+import {
+	Position
+} from 'vscode-languageserver/node';
+
+import { setTimeout } from 'timers/promises';
+
+import { FStar, FStarConfig } from './fstar';
+import { isProtocolInfo, IdeProgress, ProtocolInfo, IdeProofState, IdeQueryResponse, IdeQueryResponseTypes, IdeSymbol, IdeError, IdeAutoCompleteResponses, FullBufferQuery, FStarRange, LookupQuery, VfsAdd, AutocompleteRequest, CancelRequest } from './fstar_messages';
+import { internalPromise, withResolvers } from './utils';
+
+// For full-buffer queries, F* chunks the buffer into fragments and responds
+// with several messages, one for each fragment until the first failing
+// fragment. The series of messages ends with a full-buffer-finished. This
+// behavior allows for displaying incremental progress while the rest of the
+// buffer is being checked. `partialResult` is the type of these partially
+// completed queries.
+//
+// TODO(klinvill): would be nice to have this be a more informative type than
+// potentially containing `undefined`. E.g. to have Continuing and Done types.
+//
+// TODO(klinvill): would also be nice to have an iterator interface for
+// partialResults (or for a wrapper type like streamResults)
+export type partialResult<T> = Promise<[T, partialResult<T> | undefined]>;
```

Comment:
```suggestion
export type StreamedResult<T> = Promise<[T, StreamedResult<T> | undefined]>;
```

Nitpick: usually types are capitalized in typescript.